tests/data/model.test.ts

```diff
@@ -0,0 +1,185 @@
+import 'reflect-metadata';
+import {IModel, Model} from "../../lib/data/model";
+
+
+describe('Model maps simple properties', () => {
+
+
+    test("#string, number, and array is properly mapped", () => {
+        class MyModel extends Model {
+            public prop1: string;
+            protected prop2: number;
+            private prop3: string[];
+        }
+
+        const data = {
+            prop1: "hello",
+            prop2: 12,
+            prop3: ['a', 'b', 'c']
+        };
+
+        const myModel = new MyModel(data);
+
+        expect(myModel.model).toEqual(data);
+    });
+
+    test("#objects are properly mapped", () => {
+
+        interface ComposedObject {
+            a: string[],
+            b: {
+                a: number,
+                b: boolean[]
+            }
+        }
+
+        class MyModel extends Model {
+            public prop1: string;
+            protected prop2: ComposedObject[];
+        }
+
+        const data = {
+            prop1: "hello",
+            prop2: [
+                {
+                    a: ['a', 'b'],
+                    b: {
+                        a: 15.2,
+                        b: [true, false, true]
+                    }
+                },
+                {
+                    a: ['c', 'd'],
+                    b: {
+                        a: -3.5,
+                        b: [false, true, true]
+                    }
+                }
+            ]
+        };
+
+        const myModel = new MyModel(data);
+
+        expect(myModel.model).toEqual(data);
+    });
+});
+
+describe("Model cannot map complex properties", () => {
+    class MyClass {
+        public prop1: string;
```

Comment:
Just for the future, the naming is pretty property naming is confusing here with nested classes